.github/workflows/yaml-lint-checker.yaml

```diff
@@ -0,0 +1,68 @@
+---
+name: Yaml Lint Release Notes Pull Request
+on:
+  pull_request:
+    paths:
+      - releases/**/release-notes/**.yaml
+  # Allow manual triggering
+  workflow_dispatch: {}
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  yaml-lint-checker:
+    name: Release Notes Check Yaml 👀
+    runs-on: ubuntu-22.04
+    if: ${{ !github.event.pull_request.draft }}
+    steps:
+      - name: Cancel Previous Actions
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
```

Comment:
do you need to fetch the entire history?

Reply:
As we edit the release notes, we usually keep the history: https://github.com/kubernetes/sig-release/pull/2542 
<img width="873" alt="image" src="https://github.com/kubernetes/sig-release/assets/15183287/1dd000fb-7bfd-4052-aa77-61e72fec14f0">

But then before we merge into the sig-release repo, we clean up everything into a single commit. 

I thought `fetch-depth: 1` only fetches the latest commit? So unless we get the entire history, the workflow will only run on a single commit (which might be fine since the last commit will have the entire diff, but makes it a little harder to identify the invalid yaml during the release notes review).